### PR TITLE
api: editoast: add rollingstock_version and rename railjson_version field

### LINKED ISSUE #4584: Add a version to rolling stock
# DESCRIPTION

add a field 'rolling_stock_version' to rolling stock

# AC 

- [ ] editoast code should be updated to take into account this update
- [ ] openapi should be updated 
- [ ] rename 'version' to 'railjson_version

### PR DESCRIPTION
**Pull request by younesschrifi**
close #4584

### CODE REVIEW

**Review by Akctarus — COMMENTED** [review]
LGTM for the front part

**Comment by multun** [comment]
I have questions:
 - my understanding is that the new version fields get incremented as the infrastructure changes. I don't think it's a good idea to call that "version" is the name is already used for something else. How about "generation"?
 - why store it inside railjson at all? For all I know, it could be external metadata, hidden inside editoast

**Comment by flomonster** [comment]
> I have questions:
> 
>     * my understanding is that the new version fields get incremented as the infrastructure changes. I don't think it's a good idea to call that "version" is the name is already used for something else. How about "generation"?

We want to rename the existing field to `railjson_version` to be consistent with the infra.

>     * why store it inside railjson at all? For all I know, it could be external metadata, hidden inside editoast

I agree, we don't need to expose this field outside editoast.

**Review by clarani — APPROVED** [review]
Front looks good to me, tested in local

File: python/api/osrd_infra/models/rolling_stock.py [hunk 1]
```diff
@@ -108,6 +108,7 @@ class RollingStock(models.Model):
         null=True,
         default=None,
     )
+    rollingstock_version = models.BigIntegerField(help_text=_("Rolling stock version"), default=0)
```
Comment (flomonster) [inline python/api/osrd_infra/models/rolling_stock.py]: We can keep the name `version` to match the infra model

File: tests/fuzzer/fuzzer.py [hunk 2]
```diff
@@ -133,6 +133,7 @@ def test_new_train(
         )
     path_id = r.json()["id"]
     schedule_payload = make_payload_schedule(scenario, path_id, rolling_stock, path_length)
+    print(schedule_payload)
```
Comment (flomonster) [inline tests/fuzzer/fuzzer.py]:
```suggestion
```

File: tests/tests/test_regressions.py [hunk 3]
```diff
@@ -21,6 +21,8 @@ def _load_infra(editoast_url, infra_id):
 
 def _pathfinding_with_payload(base_url, payload, infra_id, accept_400):
     payload["infra"] = infra_id
+    print("ok")
+    print(payload)
```
Comment (flomonster) [inline tests/tests/test_regressions.py]:
```suggestion
```

File: core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingRoutesEndpoint.java [hunk 4]
```diff
@@ -68,6 +68,7 @@ public Response act(Request req) {
 
             // load rolling stocks
             var rollingStocks = List.<RollingStock>of();
+
```
Comment (flomonster) [inline core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingRoutesEndpoint.java]:
```suggestion
```

**Review by flomonster — CHANGES_REQUESTED** [review]
Thanks for the PR.
I didn't tested it yet (waiting for the rename)

**Review by flomonster — APPROVED** [review]
LGTM (tested)

**Review by Tguisnet — APPROVED** [review]
LGTM

**Review by eckter — APPROVED** [review]

**Pull request merged**